arroyo-api/src/pipelines.rs

```diff
@@ -339,35 +344,112 @@ pub async fn validate_pipeline(
+        Ok((mut program, _)) => {
+            optimizations::optimize(&mut program.graph);
+            let nodes = program
+                .graph
+                .node_weights()
+                .map(|node| PipelineNode {
+                    node_id: node.operator_id.to_string(),
+                    operator: format!("{:?}", node),
+                    parallelism: node.clone().parallelism as u32,
+                })
+                .collect();
+
+            let edges = program
+                .graph
+                .edge_references()
+                .map(|edge| {
+                    let src = program.graph.node_weight(edge.source()).unwrap();
+                    let target = program.graph.node_weight(edge.target()).unwrap();
+                    PipelineEdge {
+                        src_id: src.operator_id.to_string(),
+                        dest_id: target.operator_id.to_string(),
+                        key_type: edge.weight().key.to_string(),
+                        value_type: edge.weight().value.to_string(),
+                        edge_type: format!("{:?}", edge.weight().typ),
+                    }
+                })
+                .collect();
 
-    let nodes = program
-        .graph
-        .node_weights()
-        .map(|node| PipelineNode {
-            node_id: node.operator_id.to_string(),
-            operator: format!("{:?}", node),
-            parallelism: node.clone().parallelism as u32,
-        })
-        .collect();
-    let edges = program
-        .graph
-        .edge_references()
-        .map(|edge| {
-            let src = program.graph.node_weight(edge.source()).unwrap();
-            let target = program.graph.node_weight(edge.target()).unwrap();
-            PipelineEdge {
-                src_id: src.operator_id.to_string(),
-                dest_id: target.operator_id.to_string(),
-                key_type: edge.weight().key.to_string(),
-                value_type: edge.weight().value.to_string(),
-                edge_type: format!("{:?}", edge.weight().typ),
+            QueryValidationResult {
+                graph: Some(PipelineGraph { nodes, edges }),
+                errors: None,
             }
+        }
+        Err(e) => QueryValidationResult {
+            graph: None,
+            errors: Some(vec![e.to_string()]),
+        },
+    };
+
+    Ok(Json(pipeline_graph_validation_result))
+}
+
+/// Validate UDFs
+#[utoipa::path(
+    post,
+    path = "/v1/pipelines/validate_udfs",
+    tag = "pipelines",
+    request_body = ValidateUdfsPost,
+    responses(
+        (status = 200, description = "Validated query", body = UdfValidationResult),
+    ),
+)]
+pub async fn validate_udfs(
+    State(state): State<AppState>,
+    bearer_auth: BearerAuth,
+    WithRejection(Json(validate_udfs_post), _): WithRejection<Json<ValidateUdfsPost>, ApiError>,
+) -> Result<Json<UdfValidationResult>, ErrorResp> {
+    let _client = client(&state.pool).await?;
```

Comment:
shuold this be removed?

Reply:
that was supposed to be for the line above `_client`

Reply:
Sure

Reply:
Yes, done.